bin/seurat_merge.Rmd

```diff
@@ -21,7 +21,12 @@ params:
 ```{r, prep_args,  message=FALSE, include=FALSE}
 # set up params
 species <- params$species
-vars_to_regress_list <- strsplit(gsub(" ", "", params$vars_to_regress), ",")[[1]]
+vars_to_regress <- params$vars_to_regress
+vars_to_regress_list <- if (is.null(vars_to_regress) || toupper(vars_to_regress) == "NULL") {
+  NULL
+} else {
+  unlist(strsplit(gsub(" ", "", vars_to_regress), ","))
+}
```

Comment:
@wong-nw here's the key logic change: the `vars_to_regress_list` will be set to either `NULL` or a character vector of variables.